packages/core/upload/tests/admin/folder.test.e2e.js

```diff
@@ -326,4 +326,192 @@ describe('Folder', () => {
       data.folders.push(folder1, folder1b);
     });
   });
+
+  describe('update', () => {
+    test('rename a folder', async () => {
+      const folder = await createFolder('folder-name', null);
+
+      const res = await rq({
+        method: 'PUT',
+        url: `/upload/folders/${folder.id}`,
+        body: {
+          name: 'new name',
+        },
+      });
+
+      expect(res.body.data).toMatchObject({
+        name: 'new name',
+        path: folder.path,
+      });
+      data.folders.push(res.body.data);
+    });
+
+    test('cannot rename a folder if duplicated', async () => {
+      const folder0 = await createFolder('folder-a-0', null);
+      const folder1 = await createFolder('folder-a-1', null);
+      const folder00 = await createFolder('folder-a-00', folder0.id);
+      data.folders.push(folder0, folder1, folder00);
+
+      const res = await rq({
+        method: 'PUT',
+        url: `/upload/folders/${folder1.id}`,
+        body: {
+          name: 'folder-a-00',
+          parent: folder0.id,
+        },
+      });
+
+      expect(res.status).toBe(400);
+      expect(res.body.error.message).toBe('name already taken');
+    });
+
+    test('cannot move a folder to a folder that does not exist', async () => {
```

Comment:
Should we also add a test to prove the fact, that you can not move a folder into one of its children (which should throw an error)?

Reply:
You're right, it's not implemented. I'll implement in another PR if that's good for you?

Reply:
Nice, thanks!